src/core/DataFile.cpp

```diff
@@ -1804,7 +1804,80 @@ void DataFile::upgrade_sampleAndHold()
 		// Correct old random wave LFO speeds
 		if (e.attribute("wave").toInt() == 6)
 		{
-			e.setAttribute("speed",0.01f);
+			e.setAttribute("speed", 0.01f);
+		}
+	}
+}
+
+
+// Change loops' filenames in <sampleclip>s
+void DataFile::upgrade_loopsRename()
+{
+	const std::vector<std::pair<QString, QString>> LOOPBPMS{
+		{"briff01", "140"},
+		{"rave_bass01", "180"},
+		{"rave_bass02", "180"},
+		{"tb303_01", "123"},
+		{"techno_bass01", "140"},
+		{"techno_bass02", "140"},
+		{"techno_synth01", "140"},
+		{"techno_synth02", "140"},
+		{"techno_synth03", "130"},
+		{"techno_synth04", "140"},
+		{"909beat01", "122"},
+		{"break01", "168"},
+		{"break02", "141"},
+		{"break03", "168"},
+		{"electro_beat01", "120"},
+		{"electro_beat02", "119"},
+		{"house_loop01", "142"},
+		{"jungle01", "168"},
+		{"rave_hihat01", "180"},
+		{"rave_hihat02", "180"},
+		{"rave_kick01", "180"},
+		{"rave_kick02", "180"},
+		{"rave_snare01", "180"},
+		{"latin_brass01", "140"},
+		{"latin_guitar01", "126"},
+		{"latin_guitar02", "140"},
+		{"latin_guitar03", "120"},
+	};
+
+	const QString BPMPREF = " - ", BPMSUFF = " BPM";
+
+	// Replace loop sample names
+	for (auto& [elem, srcAttrs] : ELEMENTS_WITH_RESOURCES)
+	{
+		auto elements = elementsByTagName(elem);
+
+		for (int i = 0; i < srcAttrs.size(); ++i)
+		{
+			auto srcAttr = srcAttrs.at(i);
+
+			for (int j = 0; j < elements.length(); ++j)
+			{
+				auto item = elements.item(j).toElement();
+
+				if (item.isNull()) { continue; }
+				if (item.hasAttribute(srcAttr))
+				{
+					for (int k = 0; k < LOOPBPMS.size(); ++k)
+					{
+						auto currentPair = LOOPBPMS.at(k);
+						auto toReplace = currentPair.first;
+						auto srcVal = item.attribute(srcAttr);
+
+						if (srcVal.contains(toReplace, Qt::CaseInsensitive))
```

Comment:
I worry that this check is too loose - we could easily end up replacing names of non-factory samples too. Can it be restricted further, to only match entire file names, and only in the factory sample directory?